### PR TITLE
Allow overriding the branch a repo uses for a book 

### PR DESCRIPTION
**Pull request by nik9000**
Allows specifying a `map_branches` to override the branch used for a
particular repo. It looks like:

```
              -
                title:      JavaScript API
                prefix:     javascript-api
                current:    7.x
                branches:   [ master, 7.x, 6.x, 5.x, legacy ]
                index:      docs/index.asciidoc
                tags:       Clients/JavaScript
                subject:    Clients
                asciidoctor: true
                sources:
                  -
                    repo:   elasticsearch-js
                    path:   docs/
                    map_branches: {legacy: 16.x}
```

### CODE REVIEW

**Comment by nik9000** [comment]
I'm finishing this off, mostly around ruby style.

**Comment by nik9000** [comment]
@olksdr, could you have a look at the perl side of this?
@estolfo, could you have a look at the ruby side of this?

File: integtest/spec/helper/repo.rb [hunk 1]
```diff
@@ -54,22 +62,17 @@ def path(source_relative_path)
   # Commit all changes to the repo.
   def commit(message)
     Dir.chdir @root do
+      init_if_needed
```
[the remark this answers is not in the document]
Comment (nik9000) [reply integtest/spec/helper/repo.rb]: Sure!

**Comment by nik9000** [comment]
Thanks for reviewing @estolfo! I've made the changes you requested. Thanks for catching all of my spelling errors! I knew about the `!` thing but was never really sure when to use it.

**Comment by estolfo** [comment]
The bang is really useful and I use it quite a bit. Let me know if you're ever considering using it and want feedback on whether it suits the scenario.

**Comment by nik9000** [comment]
Thanks for reviewing @estolfo and @olksdr!

**Pull request merged**